content/en/docs/Reference/configuration.md

```diff
@@ -1034,6 +1034,51 @@ stages:
              filesystem: "ext4"
 ```
 
+You can also set custom partitions within the `kairos-install.pre.before` stage. In the following example we will do a
+custom partition in disk `/dev/vda`.
+
+
+{{% alert title="Warning" %}}
+You're responsible to make sure the sizes of the partitions fit properly within the disk. Issues of space will be highlighted by
+the agent, but they will not fail the installation process unless you pass the `--strict` flag.
+{{% /alert %}}
+
+```shell
+#cloud-config
+
+install:
+  no-format: true
+
+stages:
+  kairos-install.pre.before:
+  # TODO: check which device exists
+  - if:  '[ -e /dev/vda ] && (kairos-agent state get boot | grep -q livecd_boot)'
```

Comment:
Originally I added this check because I used the `boot` stage. I wanted to avoid partitioning the disk every time the system booted. Since then, I changed to this `kairos-install.pre.before` stage which I think is safe to run without a condition. We can keep the check for the disk though. E.g. maybe the user uses the same config on 2 different systems where the disk may be named differently.

Reply:
ok will adapt